json-schema/json-schema/src/main/scala/endpoints/algebra/JsonSchemas.scala

```diff
@@ -82,6 +82,9 @@ trait JsonSchemas {
   /** The JSON schema of a record with a single optional field `name` of type `A` */
   def optField[A](name: String, documentation: Option[String] = None)(implicit tpe: JsonSchema[A]): Record[Option[A]]
 
+  /** Name of the field that acts as discriminator for sum types (coproducts) */
+  def discriminatorName: String = "type"
+
   /** Tags a schema for type `A` with the given tag name */
   def taggedRecord[A](recordA: Record[A], tag: String): Tagged[A]
```

Comment:
What do you think of taking the `discriminatorName` as a parameter of `taggedRecord` (with a default value to `"type"`)? I don’t know if this is useful but that would allow users to have schemas with different discriminator names in a same project.

Reply:
I agree it would be useful to have control on discriminator field name per concrete union type, but I don't think it should be a property of `taggedRecord`, which reflects to coproduct instance. You could have two tagged records with different discriminator names and when summing them with `choiceTagged` we would need to make arbitrary choice which discriminator name to use which would be misleading.

A bit better would be to extend `choiceTagged` signature to:
```scala
def choiceTagged[A, B](taggedA: Tagged[A], taggedB: Tagged[B], discriminator: String = "type"): Tagged[Either[A, B]]
```
Not perfect, but at least we would have clear semantics: discriminator from the outermost `choiceTagged` is used.

Alternatively, we could introduce another combinator to the algebra, something similar to already existing `named`:

```scala
def withDiscriminator[A](tagged: Tagged[A], discriminator: String): Tagged[A]
```
which could be used for providing own discriminator field name (by default we would use just `"type"`).

WDYT?

Reply:
> Alternatively, we could introduce another combinator to the algebra, something similar to already existing `named`:
> 
> ```scala
> def withDiscriminator[A](tagged: Tagged[A], discriminator: String): Tagged[A]
> ```
> 
> which could be used for providing own discriminator field name (by default we would use just `"type"`).

Indeed, the `taggedRecord` method is not the right place to put this information. Your solution seems better.

Reply:
Implemented ✅